### PR TITLE
Ease using `make test` with Docker and improve related documentation

### PR DESCRIPTION
**Pull request by Martchus**
See particular commit messages

### CODE REVIEW

File: docs/Contributing.asciidoc [hunk 1]
```diff
@@ -667,32 +669,41 @@ Script `.circleci/build_dependencies.sh` can be also modified when major
...
+One way is to build an image using `build_local_docker.sh` script, start a
+container and then use the same commands one would use to test locally.
 
+Pull the latest base image (otherwise it may be outdated):
 ```
-# Optionally pull recent base image, otherwise it may be outdated
 docker pull registry.opensuse.org/devel/openqa/ci/containers/base:latest
 ```
 
+Create an image called `localtest` based on the content of `dependnencies.txt`
```
Comment (kalikiana) [inline docs/Contributing.asciidoc:683]:
```suggestion
Create an image called `localtest` based on the contents of `dependencies.txt`
```

File: docs/Contributing.asciidoc [hunk 2]
```diff
@@ -667,32 +669,41 @@ Script `.circleci/build_dependencies.sh` can be also modified when major
 changes are performed, e.g. different OS version or packages from forked OBS
 project, etc.
 
-=== Run tests locally using docker
+[[circleci-local-docker]]
+=== Run tests locally using Docker
 
-One way is to build image using build_local_docker.sh script, start container
-and then use the same commands one would use to test locally.
+One way is to build an image using `build_local_docker.sh` script, start a
```
Comment (kalikiana) [inline docs/Contributing.asciidoc:675]:
```suggestion
One way is to build an image using the `build_local_docker.sh` script, start a
```

File: docs/Contributing.asciidoc [hunk 3]
```diff
@@ -667,32 +669,41 @@ Script `.circleci/build_dependencies.sh` can be also modified when major
...
-docker run --rm --name t1 -v $(pwd):/opt/testing_area localtest tail -f /dev/null & sleep 1
+docker run --rm --name t1 -v $PWD:/opt/testing_area localtest tail -f /dev/null & sleep 1
 docker exec -it t1 bash -c 'make test TESTS=t/ui/25-developer_mode.t'
 docker stop -t 0 t1
 ```
 
-=== Run tests using circleci tool
+=== Run tests using the circleci tool
+
+After installing the `circleci` tool following commands will be available.
```
Comment (kalikiana) [inline docs/Contributing.asciidoc:704]:
```suggestion
After installing the `circleci` tool the following commands will be available.
```

File: docs/Contributing.asciidoc [hunk 4]
```diff
@@ -702,12 +713,12 @@ circleci local execute --job testdeveloperfullstack
 
 === Changing config.cnf
 
-Command to verify yaml with circleci tool
+Command to verify yaml with the `circleci` tool:
```
Comment (kalikiana) [inline docs/Contributing.asciidoc:716]:
```suggestion
Command to verify the YAML with the `circleci` tool:
```

**Comment by AdamWill** [comment]
"Use relative path in test-with-database target" is breaking tests in Fedora's package build environment, it seems like:

    Can't locate OpenQA/Test/PatchDeparse.pm in @INC (you may need to install the OpenQA::Test::PatchDeparse module) (@INC contains: t/lib /builddir/build/BUILD/openQA-e9b44740c89522368b2561bf94a34d86d11eb62d/lib /usr/local/lib64/perl5/5.32 /usr/local/share/perl5/5.32 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5).
    BEGIN failed--compilation aborted.
    Can't locate OpenQA/Test/PatchDeparse.pm in @INC (you may need to install the OpenQA::Test::PatchDeparse module) (@INC contains: t/lib /builddir/build/BUILD/openQA-e9b44740c89522368b2561bf94a34d86d11eb62d/lib /usr/local/lib64/perl5/5.32 /usr/local/share/perl5/5.32 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5).
    BEGIN failed--compilation aborted.
    Can't locate OpenQA/Test/PatchDeparse.pm in @INC (you may need to install the OpenQA::Test::PatchDeparse module) (@INC contains: t/lib /builddir/build/BUILD/openQA-e9b44740c89522368b2561bf94a34d86d11eb62d/lib /usr/local/lib64/perl5/5.32 /usr/local/share/perl5/5.32 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5).
    BEGIN failed--compilation aborted.
    Can't locate OpenQA/Test/PatchDeparse.pm in @INC (you may need to install the OpenQA::Test::PatchDeparse module) (@INC contains: t/lib /builddir/build/BUILD/openQA-e9b44740c89522368b2561bf94a34d86d11eb62d/lib /usr/local/lib64/perl5/5.32 /usr/local/share/perl5/5.32 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5).
    BEGIN failed--compilation aborted.
    Can't locate OpenQA/Test/PatchDeparse.pm in @INC (you may need to install the OpenQA::Test::PatchDeparse module) (@INC contains: t/lib /builddir/build/BUILD/openQA-e9b44740c89522368b2561bf94a34d86d11eb62d/lib /usr/local/lib64/perl5/5.32 /usr/local/share/perl5/5.32 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5).
    BEGIN failed--compilation aborted.

**Comment by AdamWill** [comment]
We run the tests from the 'right' directory, so I don't know why that doesn't work, but...it doesn't work.

**Comment by kalikiana** [comment]
> We run the tests from the 'right' directory, so I don't know why that doesn't work, but...it doesn't work.
> 
> edit: confirmed that commit is definitely the problem, reverting it makes the problem go away.

Do we havea  ticket or follow-up PR yet?

**Comment by Martchus** [comment]
The "right" directory is the top-level directory of the repository. That should work despite this change. The previous version wasn't working within Docker so I don't like to just revert the commit.

**Comment by AdamWill** [comment]
@Martchus that's where we run from.

**Comment by Martchus** [comment]
I've just tested it locally and it works. The openSUSE package build as well. I don't know your environment so I'm not sure what's going wrong there. The module `OpenQA::Test::PatchDeparse` is within `t/lib` so I don't see why it wouldn't work when executed within the repository root directory.

**Comment by AdamWill** [comment]
I mean, the obvious, dumb thing we could do is just have both :) `-I$(PWD)/t/lib -It/lib` or whatever the correct syntax is. You'd think that would be fine. I am curious about why the second form doesn't work, though, I can't see any immediate reason for it. Part of me wants to go find out and part of me wonders if it's the best use of my time...

**Comment by okurz** [comment]
I have seen similar issues like these with just the relative lib path not being enough. I guess I would also add both paths

**Comment by AdamWill** [comment]
I'm poking a bit at the question of 'why', but can't figure it out yet. If I manually go into the mock (the Fedora build tool) chroot and run some test commands from the same directory we run the tests from...it works OK. It's probably something really deeply weird.

**Pull request merged**